Experimental: hook version of reportWebVitals (#28769)
vercel/next.js: docs/advanced-features/measuring-performance.md

## Build Your Own

> There're 2 ways of defining web vitals metrics reporting handler.

Reviewer: ```suggestion
There're 2 ways to define your metrics handler.
```

> **First one**, you can create a [custom App](/docs/advanced-features/custom-app.md) component and define a `reportWebVitals` function as global web vitals metrics handler:

Reviewer: ```suggestion
One way is to create a [custom App](/docs/advanced-features/custom-app.md) component and define a `reportWebVitals` function:
```

export default MyApp
```

> **Another way** is to use the React hook version of `reportWebVitals`, using `useWebVitalsReport` and passing a callback to receive web vital metrics:

Reviewer: ```suggestion
Alternatively you can use the `useWebVitalsReport` hook API, with a callback function to receive metric reports:
```

```js
// pages/home.js
import { useWebVitalsReport } from 'next/vitals'

export default function Home() {
  useWebVitalsReport((metric) => {
    console.log(metric)
  })

  return <div>Home Page</div>
}
```

> The exported `reportWebVitals` function or the callback parameter of `useWebVitalsReport` is fired when the final values for any of the metrics have finished calculating on

Reviewer 2: > final

I'm not sure this is strictly true -- e.g. `web-vitals` will trigger CLS again if you close the page.

Author: replaced it with "essential"
